utils/ioutil.go

```diff
@@ -38,6 +42,8 @@ type IoUtil interface {
 	Unzip(arch string, dest string) error
 	Untar(tarPath string, dest string) error
 	CreateFile(file string, tr io.Reader) error
+	FetchJSON(url string) ([]byte, error)
+	GetPluginMetas(plugins []model.PluginFQN, defaultRegistry string) ([]model.PluginMeta, error)
```

Comment:
Other methods in this file are business logic agnostic. Do you think we need to change the approach?

Reply:
I agree about business logic in `ioutil` to an extent, but the method is fairly coupled with `fetchJSON`, which is definitely a `ioutil` method. Moving it elsewhere would require creating a `ioutil` instance, which I think is somewhat clunky. 

I'm open to suggestions though.

Reply:
Also, at first glance, moving it out would make testing trickier, since mocking http.Client is necessary.

Reply:
We need to apply separation of concerns here. Util here is supposed to sugar cover boilerplate IO operations. Dealing with meta specific stuff is not supposed to be IO boilerplate code.
Thus I suggest moving code of downloading metas (or at least parsing them since we have download IO util function) to the broker. 
main.go of unified broker now also contains business logic and it wasn't supposed to.
Instead of that I would create `StartWithFQNs` function in addition to `Start` one and move business logic there. This should also help in avoiding changing other brokers (and you can avoid removing them as well).
Taking this into account main.go would do:
```go
if cfg.DownloadMetas {
    pluginFQNs, err := cfg.ParsePluginFQNs()
    if err ..... {}
    StartWithFQNs(pluginFQNs)
} else {
    pluginMetas, err := cfg.ReadConfig()
    if err ..... {}
    Start(pluginMetas)
}
```

Since we are going to remove `Start` function or make it do what `StartWithFQNs` does we should also deprecate it.

As for the unit testing VS Code broker tests contains cases with fetching something from remote server when remote server response mocking is needed. 
Here is function for that https://github.com/eclipse/che-plugin-broker/blob/b31e49ca87c87de4c251c7b0ffa6822682213612/brokers/test/utils.go#L139 that is used here https://github.com/eclipse/che-plugin-broker/blob/1451e5d1156bff4addb639050e848bd85969baca/brokers/vscode/broker_test.go#L65 in conjunction with https://github.com/eclipse/che-plugin-broker/blob/1451e5d1156bff4addb639050e848bd85969baca/brokers/vscode/broker_test.go#L877

Reply:
:+1: Makes sense, will do.